apache2/msc_status_engine.c

```diff
@@ -111,7 +111,7 @@ int DSOLOCAL msc_status_engine_fill_with_dots(char *encoded_with_dots,
         goto return_length;
     }
 
-    for (i = 0; i < strlen(data) && i < len; i++) {
+    for (i = 0; data[i] && i < len; i++) {
```

Comment:
Consider reversing this to make it `i<len && data[i]`. That way, we prevent a segfault, just in case len is incorrect.